Send a request to CityBikes for the city of your choice.

In [13]:
import os
import requests
import json
import pandas as pd

In [14]:
request_url = 'https://api.citybik.es/v2/networks'
header_dict = {'accept': 'application/JSON'}
response = requests.get(request_url, headers=header_dict)
response

#store response in a variable
citybikes = response.json()

print(citybikes)

{'networks': [{'company': ['ЗАО «СитиБайк»'], 'href': '/v2/networks/velobike-moscow', 'id': 'velobike-moscow', 'location': {'city': 'Moscow', 'country': 'RU', 'latitude': 55.75, 'longitude': 37.616667}, 'name': 'Velobike'}, {'company': ['Urban Infrastructure Partner'], 'href': '/v2/networks/baerum-bysykkel', 'id': 'baerum-bysykkel', 'location': {'city': 'Bærum', 'country': 'NO', 'latitude': 59.89455, 'longitude': 10.546343}, 'name': 'Bysykkel'}, {'company': ['Comunicare S.r.l.'], 'href': '/v2/networks/bicincitta-siena', 'id': 'bicincitta-siena', 'location': {'city': 'Siena', 'country': 'IT', 'latitude': 43.3186, 'longitude': 11.3306}, 'name': 'Bicincittà', 'source': 'https://www.bicincitta.com/frmLeStazioni.aspx?ID=202'}, {'company': ['Cyclopolis Systems'], 'href': '/v2/networks/cyclopolis-maroussi', 'id': 'cyclopolis-maroussi', 'location': {'city': 'Maroussi', 'country': 'GR', 'latitude': 38.0568722388, 'longitude': 23.8083299536}, 'name': 'Cyclopolis'}, {'company': ['Groundwork', 'Sl

In [15]:
citybikes

{'networks': [{'company': ['ЗАО «СитиБайк»'],
   'href': '/v2/networks/velobike-moscow',
   'id': 'velobike-moscow',
   'location': {'city': 'Moscow',
    'country': 'RU',
    'latitude': 55.75,
    'longitude': 37.616667},
   'name': 'Velobike'},
  {'company': ['Urban Infrastructure Partner'],
   'href': '/v2/networks/baerum-bysykkel',
   'id': 'baerum-bysykkel',
   'location': {'city': 'Bærum',
    'country': 'NO',
    'latitude': 59.89455,
    'longitude': 10.546343},
   'name': 'Bysykkel'},
  {'company': ['Comunicare S.r.l.'],
   'href': '/v2/networks/bicincitta-siena',
   'id': 'bicincitta-siena',
   'location': {'city': 'Siena',
    'country': 'IT',
    'latitude': 43.3186,
    'longitude': 11.3306},
   'name': 'Bicincittà',
   'source': 'https://www.bicincitta.com/frmLeStazioni.aspx?ID=202'},
  {'company': ['Cyclopolis Systems'],
   'href': '/v2/networks/cyclopolis-maroussi',
   'id': 'cyclopolis-maroussi',
   'location': {'city': 'Maroussi',
    'country': 'GR',
    'latitude':

In [16]:
#print JSON response containing bike stations for Melbourne
city_id = "austin"
city_url = f'https://api.citybik.es/v2/networks/{city_id}'

response = requests.get(city_url)
austin_bikes_json= response.json()

print(austin_bikes_json)


{'network': {'company': ['BCycle, LLC'], 'gbfs_href': 'https://gbfs.bcycle.com/bcycle_austin/gbfs.json', 'href': '/v2/networks/austin', 'id': 'austin', 'location': {'city': 'Austin, TX', 'country': 'US', 'latitude': 30.26408, 'longitude': -97.74355}, 'name': 'Austin B-cycle', 'stations': [{'empty_slots': 4, 'extra': {'address': '400 Haywood Ave.', 'last_updated': 1688681065, 'renting': 1, 'returning': 1, 'uid': 'bcycle_austin_7186'}, 'free_bikes': 9, 'id': '01eedbec47378a4407f5a3fc4352fdf3', 'latitude': 30.25759, 'longitude': -97.74898, 'name': 'One Texas Center', 'timestamp': '2023-07-06T22:04:26.007000Z'}, {'empty_slots': 9, 'extra': {'address': '2711 Rio Grande Street', 'last_updated': 1688681065, 'renting': 1, 'returning': 1, 'uid': 'bcycle_austin_7189'}, 'free_bikes': 1, 'id': '57eb8e5d401aeabe67ff95e6f1f1f999', 'latitude': 30.29312, 'longitude': -97.74412, 'name': '28th/Rio Grande', 'timestamp': '2023-07-06T22:04:26.010000Z'}, {'empty_slots': 13, 'extra': {'address': '900 Pearl S

In [20]:
austin_bikes_json["network"]

{'company': ['BCycle, LLC'],
 'gbfs_href': 'https://gbfs.bcycle.com/bcycle_austin/gbfs.json',
 'href': '/v2/networks/austin',
 'id': 'austin',
 'location': {'city': 'Austin, TX',
  'country': 'US',
  'latitude': 30.26408,
  'longitude': -97.74355},
 'name': 'Austin B-cycle',
 'stations': [{'empty_slots': 4,
   'extra': {'address': '400 Haywood Ave.',
    'last_updated': 1688681065,
    'renting': 1,
    'returning': 1,
    'uid': 'bcycle_austin_7186'},
   'free_bikes': 9,
   'id': '01eedbec47378a4407f5a3fc4352fdf3',
   'latitude': 30.25759,
   'longitude': -97.74898,
   'name': 'One Texas Center',
   'timestamp': '2023-07-06T22:04:26.007000Z'},
  {'empty_slots': 9,
   'extra': {'address': '2711 Rio Grande Street',
    'last_updated': 1688681065,
    'renting': 1,
    'returning': 1,
    'uid': 'bcycle_austin_7189'},
   'free_bikes': 1,
   'id': '57eb8e5d401aeabe67ff95e6f1f1f999',
   'latitude': 30.29312,
   'longitude': -97.74412,
   'name': '28th/Rio Grande',
   'timestamp': '2023-07-

In [22]:
austin_bike_station = austin_bikes_json['network']['stations']

for station in austin_bike_station:
    station_name = station['name']
    latitude = station['latitude']
    longitude = station['longitude']
    num_bikes = station['free_bikes']
    station_id = station['id']
    
    
    print(f"station_name: {station_name}, station_id: {station_id}, Latitude: {latitude}, Longitude: {longitude}, Number of Bikes: {num_bikes}",)


station_name: One Texas Center, station_id: 01eedbec47378a4407f5a3fc4352fdf3, Latitude: 30.25759, Longitude: -97.74898, Number of Bikes: 9
station_name: 28th/Rio Grande, station_id: 57eb8e5d401aeabe67ff95e6f1f1f999, Latitude: 30.29312, Longitude: -97.74412, Number of Bikes: 1
station_name: 22nd/Pearl, station_id: ed95d8e12ad7c2ec3ff0b7dd0f4673f2, Latitude: 30.2854, Longitude: -97.74673, Number of Bikes: 0
station_name: 9th/Henderson, station_id: 8723bfa08ec83b133f6a9aeecd075f51, Latitude: 30.27217, Longitude: -97.75246, Number of Bikes: 6
station_name: South Congress @ Bouldin Creek, station_id: 568256d098257d6e8433513638f5257b, Latitude: 30.25495, Longitude: -97.74755, Number of Bikes: 4
station_name: South Congress/Mary, station_id: a22287cc0ed65baf5d88b04916ad3f8e, Latitude: 30.24496, Longitude: -97.75127, Number of Bikes: 10
station_name: 4th/Guadalupe @ Republic Square, station_id: 0071a519298ba811d05c62fcf03c39c6, Latitude: 30.26727, Longitude: -97.74718, Number of Bikes: 5
stati

In [23]:
austin_dict = {'name': [], 'station_id': [], 'Latitude': [], 'Longitude': [], 'Number of Bikes': []}

for station in austin_bike_station:
    name = station['name']
    station_id = station['id']
    latitude = station['latitude']
    longitude = station['longitude']
    num_bikes = station['free_bikes']
    
    austin_dict['name'].append(name)
    austin_dict['station_id'].append(station_id)
    austin_dict['Latitude'].append(latitude)
    austin_dict['Longitude'].append(longitude)
    austin_dict['Number of Bikes'].append(num_bikes)


In [25]:
austinbikes_df = pd.DataFrame(austin_dict)

In [26]:
austinbikes_df

name                        station_id  \
0                       One Texas Center  01eedbec47378a4407f5a3fc4352fdf3   
1                        28th/Rio Grande  57eb8e5d401aeabe67ff95e6f1f1f999   
2                             22nd/Pearl  ed95d8e12ad7c2ec3ff0b7dd0f4673f2   
3                          9th/Henderson  8723bfa08ec83b133f6a9aeecd075f51   
4         South Congress @ Bouldin Creek  568256d098257d6e8433513638f5257b   
..                                   ...                               ...   
74  East 5th/Broadway @ Capital Metro HQ  fe9c5e51f706fdd8981ed72ebf05eab4   
75                              6th/West  3c8c0510f41de2f9d61a1d3ed5993790   
76                          4th/Congress  15c906ebd3955b31efa9a1dd4de8f950   
77     11th/Congress @ The Texas Capitol  49abfbfe3cddcd1574965bef3772db3d   
78   East 5th/Shady @ Eastside Bus Plaza  37f6489cfb4ee3ffda99e30e2c84e08c   

    Latitude  Longitude  Number of Bikes  
0   30.25759  -97.74898                9  
1   30.29312  -97.74412                1  
2   30.28540  -97.74673                0  
3   30.27217  -97.75246                6  
4   30.25495  -97.74755                4  
..       ...        ...              ...  
74  30.25630  -97.71007                4  
75  30.27041  -97.75046                6  
76  30.26634  -97.74378                3  
77  30.27260  -97.74127                5  
78  30.25212  -97.69807                6  

[79 rows x 5 columns]

# Foursquare

Send a request to Foursquare with a small radius (1000m) for all the bike stations in your city of choice. 

In [27]:
import os
import requests

In [28]:
#set the keys
#foursquare_id = os.environ["FOURSQUAREAPIID"] 
#foursquare_secret = os.environ["FOURSQUAREAPISECRET"]
foursquare_api_key = os.environ['FOURSQUAREAPIKEY2']
foursquare_api_key


'fsq3JQKwJl7WkF9UliuPny8nyJNMro/pGp/qkRoYHXvZOE0='

In [29]:
URL = "https://api.foursquare.com/v3/places/search" 

params = {
            "query": "restaurant",
            "ll" : "30.25759,-97.74898",
            "open_now": "true",
            "radius": "1000"
        }
            
header_dict = {
    "Accept": "application/JSON",
    "Authorization": foursquare_api_key 
}
response = requests.request("GET", URL, params=params, headers= header_dict)
print(response.text)

{"results":[{"fsq_id":"4a4a4d21f964a520abab1fe3","categories":[{"id":13031,"name":"Burger Joint","icon":{"prefix":"https://ss3.4sqi.net/img/categories_v2/food/burger_","suffix":".png"}},{"id":13068,"name":"American Restaurant","icon":{"prefix":"https://ss3.4sqi.net/img/categories_v2/food/default_","suffix":".png"}},{"id":13145,"name":"Fast Food Restaurant","icon":{"prefix":"https://ss3.4sqi.net/img/categories_v2/food/fastfood_","suffix":".png"}}],"chains":[],"distance":213,"geocodes":{"drop_off":{"latitude":30.258682,"longitude":-97.750966},"main":{"latitude":30.258428,"longitude":-97.751056},"roof":{"latitude":30.258428,"longitude":-97.751056}},"link":"/v3/places/4a4a4d21f964a520abab1fe3","location":{"address":"603 Barton Springs Rd","census_block":"484530013111009","country":"US","cross_street":"at S. 1st St.","dma":"Austin","formatted_address":"603 Barton Springs Rd (at S. 1st St.), Austin, TX 78704","locality":"Austin","postcode":"78704","region":"TX"},"name":"Sandy's Hamburgers","

In [ ]:
res_json1 =response.json()
res_json1

{'results': [{'fsq_id': '4a4a4d21f964a520abab1fe3',
   'categories': [{'id': 13031,
     'name': 'Burger Joint',
     'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/burger_',
      'suffix': '.png'}},
    {'id': 13068,
     'name': 'American Restaurant',
     'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/default_',
      'suffix': '.png'}},
    {'id': 13145,
     'name': 'Fast Food Restaurant',
     'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/fastfood_',
      'suffix': '.png'}}],
   'chains': [],
   'distance': 213,
   'geocodes': {'drop_off': {'latitude': 30.258682, 'longitude': -97.750966},
    'main': {'latitude': 30.258428, 'longitude': -97.751056},
    'roof': {'latitude': 30.258428, 'longitude': -97.751056}},
   'link': '/v3/places/4a4a4d21f964a520abab1fe3',
   'location': {'address': '603 Barton Springs Rd',
    'census_block': '484530013111009',
    'country': 'US',
    'cross_street': 'at S. 1st St.',
    'dma': 'Austin',

Parse through the response to get the POI (such as restaurants, bars, etc) details you want (ratings, name, location, etc)

In [36]:
res_json1["results"]

[{'fsq_id': '4a4a4d21f964a520abab1fe3',
  'categories': [{'id': 13031,
    'name': 'Burger Joint',
    'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/burger_',
     'suffix': '.png'}},
   {'id': 13068,
    'name': 'American Restaurant',
    'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/default_',
     'suffix': '.png'}},
   {'id': 13145,
    'name': 'Fast Food Restaurant',
    'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/fastfood_',
     'suffix': '.png'}}],
  'chains': [],
  'distance': 213,
  'geocodes': {'drop_off': {'latitude': 30.258682, 'longitude': -97.750966},
   'main': {'latitude': 30.258428, 'longitude': -97.751056},
   'roof': {'latitude': 30.258428, 'longitude': -97.751056}},
  'link': '/v3/places/4a4a4d21f964a520abab1fe3',
  'location': {'address': '603 Barton Springs Rd',
   'census_block': '484530013111009',
   'country': 'US',
   'cross_street': 'at S. 1st St.',
   'dma': 'Austin',
   'formatted_address': '603 Barto

In [37]:
# for citybikes DataFtame 'austinbikes_df' with station data
# query foursquareAPi for data on austinbikes 
foursquare_api_key = os.environ['FOURSQUAREAPIKEY2']
foursquare_api_key


URL = "https://api.foursquare.com/v3/places/search"

header_dict = {
    "Accept": "application/json",
    "Authorization": foursquare_api_key
}

for index, row in austinbikes_df.iterrows():
    # Extract station coordinates
    lat = row['Latitude']
    lng = row['Longitude']
    station_id = row['station_id']
    location = f"{lat},{lng}"
    station = f"{station_id}"

    # Set parameters for Foursquare API request
    params = {
        "query": "restaurant",
        "ll": location,
        "radius": "1000",
        "Limit": 20
    }
            
    response = requests.request("GET", URL, params=params, headers= header_dict)
    res_json2 = response.json()

In [38]:
res_json2

{'results': [{'fsq_id': '4aa83af3f964a520415020e3',
   'categories': [{'id': 13148,
     'name': 'French Restaurant',
     'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/french_',
      'suffix': '.png'}}],
   'chains': [],
   'distance': 264,
   'geocodes': {'main': {'latitude': 30.253272, 'longitude': -97.700578},
    'roof': {'latitude': 30.253272, 'longitude': -97.700578}},
   'link': '/v3/places/4aa83af3f964a520415020e3',
   'location': {'address': '4710 E 5th St',
    'census_block': '484530021112024',
    'country': 'US',
    'dma': 'Austin',
    'formatted_address': '4710 E 5th St, Austin, TX 78702',
    'locality': 'Austin',
    'postcode': '78702',
    'region': 'TX'},
   'name': "Justine's Brasserie",
   'related_places': {},
   'timezone': 'America/Chicago'},
  {'fsq_id': '53075b12498e717329657cd5',
   'categories': [{'id': 13049,
     'name': 'Diner',
     'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/food/diner_',
      'suffix': '.png'}},
  

In [39]:
for i in res_json2["results"]:
    for category in i["categories"]:
        print(category["name"])
    for chain in i["chains"]:
        print(chains['id'])

French Restaurant
Diner
American Restaurant
Cajun and Creole Restaurant
Cocktail Bar
Wine Bar
Taco Restaurant
Food Truck
BBQ Joint
Mexican Restaurant
Tex-Mex Restaurant
Fruit and Vegetable Store
Grocery Store
Burger Joint
American Restaurant
Fast Food Restaurant
American Restaurant
New American Restaurant


Put your parsed results into a DataFrame

In [30]:
import pandas as pd
import requests

# Assuming you have a dataframe called 'austinbikes_df' with station data
foursquare_api_key = os.environ['FOURSQUAREAPIKEY2']

URL = "https://api.foursquare.com/v3/places/search"

header_dict = {
    "Accept": "application/json",
    "Authorization": foursquare_api_key
}

a = []
b = []
c = []
d = []
e = []
f = []
g = []

for index, row in austinbikes_df.iterrows():
    # Extract station coordinates
    lat = row['Latitude']
    lng = row['Longitude']
    location = f"{lat},{lng}"
    station_id = row['station_id']

    # Set parameters for Foursquare API request
    params = {
        "query": "restaurant",
        "ll": location,
        "radius": "1000"
    }

    response = requests.get(URL, params=params, headers=header_dict)
    res_json = response.json()
    try:
        for i in res_json["results"]:
            a.append(i.get("name", None))  # Append name to 'a' list
            if "address" in i.get("location", {}):
                b.append(i["location"]["address"])  # Append address to 'b' list
            else:
                b.append(None)
            if "categories" in i:
                c.append(i["categories"][0].get("id", None))  # Append category ID to 'c' list
                d.append(i["categories"][0].get("name", None))  # Append category name to 'd' list
            else:
                c.append(None)
                d.append(None)
            e.append(i["location"].get("postcode", None))  # Append postcode to 'e' list
            f.append(i.get("distance", None))  # Append distance to 'f' list
            g.append(station_id)
    except:
        pass

foursquare_df = pd.DataFrame({"name": a, "address": b, "category_id": c, "category_name": d, "postcode": e, "distance": f, "station_id": g})

# Print or do further processing with the dataframe 'df'
print(foursquare_df)

                                         name                 address  \
0                          Sandy's Hamburgers   603 Barton Springs Rd   
1                     Freebirds World Burrito      515 S Congress Ave   
2                                  Taco Joint      134 E Riverside Dr   
3                                 Whataburger   601 Barton Springs Rd   
4                                  Aba Austin     1011 S Congress Ave   
..                                        ...                     ...   
773                 Beto's Mexican Restaurant           3518 E 7th St   
774  El Secreto Dela Abuela Mexican Restauran        817 Airport Blvd   
775                   Segovia Produce Company           4618 E 7th St   
776                 Dan's Hamburgers - Austin        844 Airport Blvd   
777          Jacoby's Restaurant & Mercantile  3235 E Cesar Chavez St   

     category_id              category_name postcode  distance  \
0          13031               Burger Joint    78704     

In [31]:
foursquare_df

name                 address  \
0                          Sandy's Hamburgers   603 Barton Springs Rd   
1                     Freebirds World Burrito      515 S Congress Ave   
2                                  Taco Joint      134 E Riverside Dr   
3                                 Whataburger   601 Barton Springs Rd   
4                                  Aba Austin     1011 S Congress Ave   
..                                        ...                     ...   
773                 Beto's Mexican Restaurant           3518 E 7th St   
774  El Secreto Dela Abuela Mexican Restauran        817 Airport Blvd   
775                   Segovia Produce Company           4618 E 7th St   
776                 Dan's Hamburgers - Austin        844 Airport Blvd   
777          Jacoby's Restaurant & Mercantile  3235 E Cesar Chavez St   

     category_id              category_name postcode  distance  \
0          13031               Burger Joint    78704       213   
1          13145       Fast Food Restaurant    78704       270   
2          13145       Fast Food Restaurant    78704       354   
3          13145       Fast Food Restaurant    78704       172   
4          13000        Dining and Drinking    78704       426   
..           ...                        ...      ...       ...   
773        13303         Mexican Restaurant    78702       651   
774        13307         Tex-Mex Restaurant    78702       715   
775        17067  Fruit and Vegetable Store    78702       500   
776        13031               Burger Joint    78702       808   
777        13068        American Restaurant    78702       891   

                           station_id  
0    01eedbec47378a4407f5a3fc4352fdf3  
1    01eedbec47378a4407f5a3fc4352fdf3  
2    01eedbec47378a4407f5a3fc4352fdf3  
3    01eedbec47378a4407f5a3fc4352fdf3  
4    01eedbec47378a4407f5a3fc4352fdf3  
..                                ...  
773  37f6489cfb4ee3ffda99e30e2c84e08c  
774  37f6489cfb4ee3ffda99e30e2c84e08c  
775  37f6489cfb4ee3ffda99e30e2c84e08c  
776  37f6489cfb4ee3ffda99e30e2c84e08c  
777  37f6489cfb4ee3ffda99e30e2c84e08c  

[778 rows x 7 columns]

In [32]:
foursquare_df.head(20)

name                address  category_id  \
0        Sandy's Hamburgers  603 Barton Springs Rd        13031   
1   Freebirds World Burrito     515 S Congress Ave        13145   
2                Taco Joint     134 E Riverside Dr        13145   
3               Whataburger  601 Barton Springs Rd        13145   
4                Aba Austin    1011 S Congress Ave        13000   
5          Thundercloud Sub     201 E Riverside Dr        13334   
6                sweetgreen    1007 S Congress Ave        13065   
7          Soco Bar & Bites     300 S Congress Ave        13009   
8                 Two Hands    1011 S Congress Ave        13034   
9           AustinMed Break     343 S Congress Ave        13000   
10           Torchy’S Tacos      2801 Guadalupe St        13145   
11               Halal Bros      2712 Guadalupe St        13191   
12     Dirty Martin's Place      2808 Guadalupe St        13031   
13      Cabo Bob's Burritos     2828 Rio Grande St        13145   
14              Whataburger      2800 Guadalupe St        13145   
15    Counter Cafe - Austin          603 W 29th St        13034   
16                   Arlo's          2908 Fruth St        13031   
17                  Sip Pho          512 W 29th St        13347   
18         Kerbey Lane Cafe      2606 Guadalupe St        13034   
19          In-N-Out Burger      2700 Guadalupe St        13031   

           category_name postcode  distance                        station_id  
0           Burger Joint    78704       213  01eedbec47378a4407f5a3fc4352fdf3  
1   Fast Food Restaurant    78704       270  01eedbec47378a4407f5a3fc4352fdf3  
2   Fast Food Restaurant    78704       354  01eedbec47378a4407f5a3fc4352fdf3  
3   Fast Food Restaurant    78704       172  01eedbec47378a4407f5a3fc4352fdf3  
4    Dining and Drinking    78704       426  01eedbec47378a4407f5a3fc4352fdf3  
5          Sandwich Spot    78704       440  01eedbec47378a4407f5a3fc4352fdf3  
6             Restaurant    78704       424  01eedbec47378a4407f5a3fc4352fdf3  
7           Cocktail Bar    78704       230  01eedbec47378a4407f5a3fc4352fdf3  
8                   Café    78704       472  01eedbec47378a4407f5a3fc4352fdf3  
9    Dining and Drinking    78704       240  01eedbec47378a4407f5a3fc4352fdf3  
10  Fast Food Restaurant    78705       238  57eb8e5d401aeabe67ff95e6f1f1f999  
11      Halal Restaurant    78705       243  57eb8e5d401aeabe67ff95e6f1f1f999  
12          Burger Joint    78705       196  57eb8e5d401aeabe67ff95e6f1f1f999  
13  Fast Food Restaurant    78705       304  57eb8e5d401aeabe67ff95e6f1f1f999  
14  Fast Food Restaurant    78705       191  57eb8e5d401aeabe67ff95e6f1f1f999  
15                  Café    78705       262  57eb8e5d401aeabe67ff95e6f1f1f999  
16          Burger Joint    78705       284  57eb8e5d401aeabe67ff95e6f1f1f999  
17      Tapas Restaurant    78705       315  57eb8e5d401aeabe67ff95e6f1f1f999  
18                  Café    78705       328  57eb8e5d401aeabe67ff95e6f1f1f999  
19          Burger Joint    78705       266  57eb8e5d401aeabe67ff95e6f1f1f999

In [40]:
foursquare_df.to_csv("foursquare_data.csv", index=False)

# Yelp

Send a request to Yelp with a small radius (1000m) for all the bike stations in your city of choice. 

In [41]:
import os
yelp_api_key = os.environ["yelp_key2"]

In [45]:
import os
import requests

yelp_api_key = os.environ['yelp_key2']

URL = "https://api.yelp.com/v3/businesses/search" 

params = {
    "term": "restaurant",
    "location": "Austin, TX",
    "open_now": "true",
    "radius": "1000"
}

header_dict = {
    "Accept": "application/JSON",
    "Authorization": "Bearer " + yelp_api_key
}

response = requests.get(URL, params=params, headers=header_dict)
print(response.text)


{"businesses": [{"id": "cIbVQrylwj3Rljpka5AZ9w", "alias": "olive-and-june-austin-2", "name": "Olive & June", "image_url": "https://s3-media2.fl.yelpcdn.com/bphoto/NrpaVA2iBUHJ6y-7UtYmiQ/o.jpg", "is_closed": false, "url": "https://www.yelp.com/biz/olive-and-june-austin-2?adjust_creative=kPz8Z0G_dj_5_xxsv6OR5A&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=kPz8Z0G_dj_5_xxsv6OR5A", "review_count": 688, "categories": [{"alias": "italian", "title": "Italian"}], "rating": 4.0, "coordinates": {"latitude": 30.30726, "longitude": -97.75113}, "transactions": ["pickup", "delivery"], "price": "$$$", "location": {"address1": "3411 Glenview Ave", "address2": "", "address3": "", "city": "Austin", "zip_code": "78703", "country": "US", "state": "TX", "display_address": ["3411 Glenview Ave", "Austin, TX 78703"]}, "phone": "+15124679898", "display_phone": "(512) 467-9898", "distance": 114.85262584680049}, {"id": "WF7rhICrqgSw7flzLvnK1A", "alias": "wu-chow-north-lamar-austin", "name

Parse through the response to get the POI (such as restaurants, bars, etc) details you want (ratings, name, location, etc)

In [47]:
yelp_json = response.json()
yelp_json
yelp_json["businesses"]

[{'id': 'cIbVQrylwj3Rljpka5AZ9w',
  'alias': 'olive-and-june-austin-2',
  'name': 'Olive & June',
  'image_url': 'https://s3-media2.fl.yelpcdn.com/bphoto/NrpaVA2iBUHJ6y-7UtYmiQ/o.jpg',
  'is_closed': False,
  'url': 'https://www.yelp.com/biz/olive-and-june-austin-2?adjust_creative=kPz8Z0G_dj_5_xxsv6OR5A&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=kPz8Z0G_dj_5_xxsv6OR5A',
  'review_count': 688,
  'categories': [{'alias': 'italian', 'title': 'Italian'}],
  'rating': 4.0,
  'coordinates': {'latitude': 30.30726, 'longitude': -97.75113},
  'transactions': ['pickup', 'delivery'],
  'price': '$$$',
  'location': {'address1': '3411 Glenview Ave',
   'address2': '',
   'address3': '',
   'city': 'Austin',
   'zip_code': '78703',
   'country': 'US',
   'state': 'TX',
   'display_address': ['3411 Glenview Ave', 'Austin, TX 78703']},
  'phone': '+15124679898',
  'display_phone': '(512) 467-9898',
  'distance': 114.85262584680049},
 {'id': 'WF7rhICrqgSw7flzLvnK1A',
  'alia

In [ ]:
for i in yelp_json["businesses"]:
   # print(i["name"]) #this prints each of the dict
    #print(i["review_count"])
    #print(i["distance"])
    #print(i["transactions"])
    #print(i["categories"][1]["alias"])
    #print(i["categories"][1]["title"])
    # print(i["location"]["region"])

SyntaxError: incomplete input (2274644815.py, line 8)

Put your parsed results into a DataFrame

In [48]:
import pandas as pd
import requests
import os

# Assuming you have a dataframe called 'austinbikes_df' with station data
yelp_key = os.environ['yelp_key2']

URL = "https://api.yelp.com/v3/businesses/search"

header_dict = {
    "Accept": "application/JSON",
    "Authorization": "Bearer " + yelp_key
}

a = []
b = []
c = []
d = []
e = []
f = []
g = []
h = []
j = []
k = []
m = []

for index, row in austinbikes_df.iterrows():
    # Extract station coordinates
    lat = row['Latitude']
    lng = row['Longitude']
    location = f"{lat},{lng}"
    station_id = row['station_id']

    # Set parameters for Yelp API request
    params = {
        "term": "restaurant",
        "location": location,
        "radius": "1000"
    }

    response = requests.get(URL, params=params, headers=header_dict)
    res_json = response.json()

    try:
        for i in res_json["businesses"]:
            a.append(i.get("name", None))  # Append name to 'a' list
            if "location" in i and "address1" in i["location"]:
                b.append(i["location"]["address1"])  # Append address to 'b' list
                j.append(i["location"]["zip_code"])
            else:
                b.append(None)
                j.append(None)
            c.append(i.get("distance", None))  # Append transactions to 'c' list
            if "categories" in i and isinstance(i["categories"], list) and len(i["categories"]) > 0:
                d.append(i["categories"][0].get("title", None))  # Append category name to 'd' list
            else:
                d.append(None)
            e.append(i.get("review_count", None))  # Append review count to 'e' list
            f.append(i.get("rating", None))  # Append rating to 'f' list
            g.append(i["coordinates"].get("longitude", None))  # Append longitude to 'g' list
            h.append(i["coordinates"].get("latitude", None))  # Append latitude to 'h' list
            m.append(i.get("price", None))
            k.append(station_id)
    except:
        pass

yelp_df = pd.DataFrame({"name": a, "address": b, "distance": c, "category_name": d, "review_count": e, "rating": f, "longitude": g, "latitude": h, "postcode": j, "station_id":k, "price": m})

# Print or do further processing with the dataframe 'yelp_df'
print(yelp_df)


                            name                  address     distance  \
0                   Aba - Austin      1011 S Congress Ave   471.988869   
1                Prim and Proper      1007 S Congress Ave   400.808357   
2                        Red Ash          303 Colorado St  1013.704865   
3     Summer House on Music Lane            1101 Music Ln   569.116145   
4                      Two Hands  1011 South Congress Ave   460.097583   
...                          ...                      ...          ...   
1523          Taqueria La Morena         5419 Glissman Rd   605.458708   
1524              Native Grocery           705 Tillery St  1165.804443   
1525                 Panda Label         5323 Levander Lp   287.846224   
1526        Pueblo Viejo Tillery           641 Tillery St   982.714120   
1527               Solar Falafel             5305 Bolm Rd   952.034772   

           category_name  review_count  rating  longitude   latitude postcode  \
0          Mediterranean      

In [49]:
yelp_df.head(20)

name                  address  \
0                                Aba - Austin      1011 S Congress Ave   
1                             Prim and Proper      1007 S Congress Ave   
2                                     Red Ash          303 Colorado St   
3                  Summer House on Music Lane            1101 Music Ln   
4                                   Two Hands  1011 South Congress Ave   
5                      Terry Black's Barbecue   1003 Barton Springs Rd   
6                          Joann’s Fine Foods      1224 S Congress Ave   
7                            Soco Bar & Bites       300 S Congress Ave   
8                           Austin Rotisserie         111 Congress Ave   
9                                 Lucky Robot      1303 S Congress Ave   
10                                 Taco Joint       134 E Riverside Dr   
11                                    El Alma   1025 Barton Springs Rd   
12                 Aussie's Grill & Beach Bar    306 Barton Springs Rd   
13  Gus's World Famous Fried Chicken - Austin     117 San Jacinto Blvd   
14                  Arlo Grey by Kristen Kish    111 E Cesar Chavez St   
15                                    Perla's      1400 S Congress Ave   
16                                 La Piscina             600 W 2nd St   
17                                 ATX Cocina       110 San Antonio St   
18                              Swift's Attic         315 Congress Ave   
19                 Lamberts Downtown Barbecue             401 W 2nd St   

       distance       category_name  review_count  rating  longitude  \
0    471.988869       Mediterranean          1086     4.5 -97.748030   
1    400.808357       Cocktail Bars            41     4.5 -97.747619   
2   1013.704865             Italian          1204     4.5 -97.744672   
3    569.116145  Breakfast & Brunch           164     4.0 -97.747695   
4    460.097583       Cocktail Bars           328     4.0 -97.747924   
5    592.378526            Barbeque          6100     4.5 -97.754801   
6    672.358172              Diners           311     4.0 -97.749150   
7    242.973635       Cocktail Bars            18     4.0 -97.747290   
8    850.685565        Comfort Food           162     4.5 -97.743576   
9    748.499198          Sushi Bars          1358     4.0 -97.749006   
10   370.641895             Mexican           563     4.5 -97.745520   
11   682.073268             Mexican          1186     4.0 -97.755620   
12   181.366866                Bars           211     3.0 -97.748080   
13   968.069424            Southern          3098     4.5 -97.741731   
14   748.641494      American (New)           398     4.0 -97.744106   
15   850.954305             Seafood          1845     4.0 -97.749930   
16   960.143650             Mexican            91     4.5 -97.749945   
17   818.912692             Mexican           637     4.0 -97.749259   
18  1065.938767      American (New)          1166     4.0 -97.743370   
19   859.187253            Barbeque          1977     4.0 -97.747905   

     latitude postcode                        station_id price  
0   30.253570    78704  01eedbec47378a4407f5a3fc4352fdf3  None  
1   30.254225    78704  01eedbec47378a4407f5a3fc4352fdf3    $$  
2   30.265880    78701  01eedbec47378a4407f5a3fc4352fdf3  $$$$  
3   30.252618    78704  01eedbec47378a4407f5a3fc4352fdf3    $$  
4   30.253579    78704  01eedbec47378a4407f5a3fc4352fdf3    $$  
5   30.259692    78704  01eedbec47378a4407f5a3fc4352fdf3    $$  
6   30.251570    78704  01eedbec47378a4407f5a3fc4352fdf3    $$  
7   30.258490    78704  01eedbec47378a4407f5a3fc4352fdf3    $$  
8   30.263536    78701  01eedbec47378a4407f5a3fc4352fdf3     $  
9   30.250851    78704  01eedbec47378a4407f5a3fc4352fdf3    $$  
10  30.256370    78704  01eedbec47378a4407f5a3fc4352fdf3     $  
11  30.260100    78704  01eedbec47378a4407f5a3fc4352fdf3    $$  
12  30.258980    78704  01eedbec47378a4407f5a3fc4352fdf3    $$  
13  30.263486    78701  01eedbec47378a4407f5a3fc4352fdf3    $$ 

In [50]:
yelp_df.to_csv('yelp_data.csv', index=False)

# Comparing Results

In [54]:
foursquare_df.describe

<bound method NDFrame.describe of                                          name                 address  \
0                          Sandy's Hamburgers   603 Barton Springs Rd   
1                     Freebirds World Burrito      515 S Congress Ave   
2                                  Taco Joint      134 E Riverside Dr   
3                                 Whataburger   601 Barton Springs Rd   
4                                  Aba Austin     1011 S Congress Ave   
..                                        ...                     ...   
773                 Beto's Mexican Restaurant           3518 E 7th St   
774  El Secreto Dela Abuela Mexican Restauran        817 Airport Blvd   
775                   Segovia Produce Company           4618 E 7th St   
776                 Dan's Hamburgers - Austin        844 Airport Blvd   
777          Jacoby's Restaurant & Mercantile  3235 E Cesar Chavez St   

     category_id              category_name postcode  distance  \
0          13031       

In [55]:
yelp_df.describe

<bound method NDFrame.describe of                             name                  address     distance  \
0                   Aba - Austin      1011 S Congress Ave   471.988869   
1                Prim and Proper      1007 S Congress Ave   400.808357   
2                        Red Ash          303 Colorado St  1013.704865   
3     Summer House on Music Lane            1101 Music Ln   569.116145   
4                      Two Hands  1011 South Congress Ave   460.097583   
...                          ...                      ...          ...   
1523          Taqueria La Morena         5419 Glissman Rd   605.458708   
1524              Native Grocery           705 Tillery St  1165.804443   
1525                 Panda Label         5323 Levander Lp   287.846224   
1526        Pueblo Viejo Tillery           641 Tillery St   982.714120   
1527               Solar Falafel             5305 Bolm Rd   952.034772   

           category_name  review_count  rating  longitude   latitude postcode

In [56]:
foursquare_df.shape

(778, 7)

In [57]:
yelp_df.shape

(1528, 11)

Clearly, the Yelp API has got a more robust data response than the foursquare APi

In [51]:
#Check for common records based on name column:
common_records = foursquare_df.merge(yelp_df, on='name', how='inner')
common_records


name               address_x  category_id  \
0                    Taco Joint      134 E Riverside Dr        13145   
1                    Taco Joint      134 E Riverside Dr        13145   
2                    Taco Joint      134 E Riverside Dr        13145   
3                    Taco Joint      134 E Riverside Dr        13145   
4                    Taco Joint      134 E Riverside Dr        13145   
...                         ...                     ...          ...   
1858                Sawyer & Co  4827 E Cesar Chavez St        13049   
1859            De Nada Cantina  4715 E Cesar Chavez St        13009   
1860            De Nada Cantina  4715 E Cesar Chavez St        13009   
1861  Beto's Mexican Restaurant           3518 E 7th St        13303   
1862  Dan's Hamburgers - Austin        844 Airport Blvd        13031   

           category_name_x postcode_x  distance_x  \
0     Fast Food Restaurant      78704         354   
1     Fast Food Restaurant      78704         354   
2     Fast Food Restaurant      78704         354   
3     Fast Food Restaurant      78704         354   
4     Fast Food Restaurant      78704         354   
...                    ...        ...         ...   
1858                 Diner      78702         367   
1859          Cocktail Bar      78702         400   
1860          Cocktail Bar      78702         400   
1861    Mexican Restaurant      78702         651   
1862          Burger Joint      78702         808   

                          station_id_x               address_y  distance_y  \
0     01eedbec47378a4407f5a3fc4352fdf3      134 E Riverside Dr  370.641895   
1     01eedbec47378a4407f5a3fc4352fdf3      134 E Riverside Dr  252.611181   
2     01eedbec47378a4407f5a3fc4352fdf3   2809 San Jacinto Blvd  274.005081   
3     01eedbec47378a4407f5a3fc4352fdf3   2809 San Jacinto Blvd  336.780584   
4     01eedbec47378a4407f5a3fc4352fdf3      134 E Riverside Dr  395.004798   
...                                ...                     ...         ...   
1858  37f6489cfb4ee3ffda99e30e2c84e08c  4827 E Cesar Chavez St  375.673199   
1859  37f6489cfb4ee3ffda99e30e2c84e08c  4715 E Cesar Chavez St  921.862744   
1860  37f6489cfb4ee3ffda99e30e2c84e08c  4715 E Cesar Chavez St  408.984337   
1861  37f6489cfb4ee3ffda99e30e2c84e08c           3518 E 7th St  674.326163   
1862  37f6489cfb4ee3ffda99e30e2c84e08c        844 Airport Blvd  807.452493   

     category_name_y  review_count  rating  longitude   latitude postcode_y  \
0            Mexican           563     4.5 -97.745520  30.256370      78704   
1            Mexican           563     4.5 -97.745520  30.256370      78704   
2            Mexican           485     4.5 -97.734965  30.291625      78705   
3            Mexican           485     4.5 -97.734965  30.291625      78705   
4            Mexican           563     4.5 -97.745520  30.256370      78704   
...              ...           ...     ...        ...        ...        ...   
1858    Cajun/Creole           895     4.0 -97.701782  30.251333      78702   
1859         Mexican           314     4.5 -97.702174  30.251447      78702   
1860         Mexican           314     4.5 -97.702174  30.251447      78702   
1861         Mexican            55     4.0 -97.702342  30.256849      78702   
1862         Burgers           166     4.0 -97.693840  30.258469      78702   

                          station_id_y price  
0     01eedbec47378a4407f5a3fc4352fdf3     $  
1     568256d098257d6e8433513638f5257b     $  
2     68f152a5c3e6e60cea8b8e1430e28906     $  
3     5d489ee85f9f1d9aeec3b596f1a0026e     $  
4     71ac94f8934969e97ab7682575088a0a     $  
...                                ...   ...  
1858  37f6489cfb4ee3ffda99e30e2c84e08c    $$  
1859  fe9c5e51f706fdd8981ed72ebf05eab4    $$  
1860  37f6489cfb4ee3ffda99e30e2c84e08c    $$  
1861  37f6489cfb4ee3ffda99e30e2c84e08c     $  
1862  37f6489cfb4ee3ffda99e30e2c84e08c     $  

[1863 rows x 17 columns]

In [52]:
#Check for common records based on name column:
common_records = foursquare_df.merge(yelp_df, on='address', how='inner')
common_records

name_x                 address  \
0                                   Taco Joint      134 E Riverside Dr   
1                                   Taco Joint      134 E Riverside Dr   
2                                   Taco Joint      134 E Riverside Dr   
3                                   Taco Joint      134 E Riverside Dr   
4                                   Taco Joint      134 E Riverside Dr   
...                                        ...                     ...   
2920                           De Nada Cantina  4715 E Cesar Chavez St   
2921                           De Nada Cantina  4715 E Cesar Chavez St   
2922                 Beto's Mexican Restaurant           3518 E 7th St   
2923  El Secreto Dela Abuela Mexican Restauran        817 Airport Blvd   
2924                 Dan's Hamburgers - Austin        844 Airport Blvd   

      category_id       category_name_x postcode_x  distance_x  \
0           13145  Fast Food Restaurant      78704         354   
1           13145  Fast Food Restaurant      78704         354   
2           13145  Fast Food Restaurant      78704         354   
3           13145  Fast Food Restaurant      78704         354   
4           13145  Fast Food Restaurant      78704         354   
...           ...                   ...        ...         ...   
2920        13009          Cocktail Bar      78702         400   
2921        13009          Cocktail Bar      78702         400   
2922        13303    Mexican Restaurant      78702         651   
2923        13307    Tex-Mex Restaurant      78702         715   
2924        13031          Burger Joint      78702         808   

                          station_id_x                     name_y  distance_y  \
0     01eedbec47378a4407f5a3fc4352fdf3                 Taco Joint  370.641895   
1     01eedbec47378a4407f5a3fc4352fdf3                 Taco Joint  252.611181   
2     01eedbec47378a4407f5a3fc4352fdf3                 Taco Joint  395.004798   
3     01eedbec47378a4407f5a3fc4352fdf3                 Taco Joint  260.673302   
4     01eedbec47378a4407f5a3fc4352fdf3                 Taco Joint  437.020613   
...                                ...                        ...         ...   
2920  37f6489cfb4ee3ffda99e30e2c84e08c            De Nada Cantina  921.862744   
2921  37f6489cfb4ee3ffda99e30e2c84e08c            De Nada Cantina  408.984337   
2922  37f6489cfb4ee3ffda99e30e2c84e08c  Beto's Mexican Restaurant  674.326163   
2923  37f6489cfb4ee3ffda99e30e2c84e08c       El Secreto De Abuela  718.756198   
2924  37f6489cfb4ee3ffda99e30e2c84e08c  Dan's Hamburgers - Austin  807.452493   

     category_name_y  review_count  rating  longitude   latitude postcode_y  \
0            Mexican           563     4.5 -97.745520  30.256370      78704   
1            Mexican           563     4.5 -97.745520  30.256370      78704   
2            Mexican           563     4.5 -97.745520  30.256370      78704   
3            Mexican           563     4.5 -97.745520  30.256370      78704   
4            Mexican           563     4.5 -97.745520  30.256370      78704   
...              ...           ...     ...        ...        ...        ...   
2920         Mexican           314     4.5 -97.702174  30.251447      78702   
2921         Mexican           314     4.5 -97.702174  30.251447      78702   
2922         Mexican            55     4.0 -97.702342  30.256849      78702   
2923         Mexican            98     4.0 -97.692479  30.256475      78702   
2924         Burgers           166     4.0 -97.693840  30.258469      78702   

                          station_id_y price  
0     01eedbec47378a4407f5a3fc4352fdf3     $  
1     568256d098257d6e8433513638f5257b     $  
2     71ac94f8934969e97ab7682575088a0a     $  
3     275ef76f31c0c1881d5ccf2880507495     $  
4     2fce3cfef72f83dcaad46ee4025cfe99     $  
...                                ...   ...  
2920  fe9c5e51f706fdd8981ed72ebf05eab4    $$  
2921  37f6489cfb4ee3ffda99e30e2c84e08c    $$  
2922  37f6489

In [53]:
#Check for records that exist in the Foursquare data frame but not in the Yelp data frame based  on name
foursquare_only = foursquare_df[~foursquare_df['name'].isin(yelp_df['name'])]
foursquare_only


name                 address  \
0                          Sandy's Hamburgers   603 Barton Springs Rd   
1                     Freebirds World Burrito      515 S Congress Ave   
3                                 Whataburger   601 Barton Springs Rd   
4                                  Aba Austin     1011 S Congress Ave   
5                            Thundercloud Sub      201 E Riverside Dr   
..                                        ...                     ...   
767                                    Eureka            200 E 6th St   
771                      Taqueria La Chilanga  5011 E Cesar Chavez St   
772                              JNL Barbecue           5012 E 7th St   
774  El Secreto Dela Abuela Mexican Restauran        817 Airport Blvd   
775                   Segovia Produce Company           4618 E 7th St   

     category_id              category_name postcode  distance  \
0          13031               Burger Joint    78704       213   
1          13145       Fast Food Restaurant    78704       270   
3          13145       Fast Food Restaurant    78704       172   
4          13000        Dining and Drinking    78704       426   
5          13334              Sandwich Spot    78704       440   
..           ...                        ...      ...       ...   
767        13031               Burger Joint    78701       531   
771        13054                 Food Truck    78702       137   
772        13026                  BBQ Joint    78702       239   
774        13307         Tex-Mex Restaurant    78702       715   
775        17067  Fruit and Vegetable Store    78702       500   

                           station_id  
0    01eedbec47378a4407f5a3fc4352fdf3  
1    01eedbec47378a4407f5a3fc4352fdf3  
3    01eedbec47378a4407f5a3fc4352fdf3  
4    01eedbec47378a4407f5a3fc4352fdf3  
5    01eedbec47378a4407f5a3fc4352fdf3  
..                                ...  
767  49abfbfe3cddcd1574965bef3772db3d  
771  37f6489cfb4ee3ffda99e30e2c84e08c  
772  37f6489cfb4ee3ffda99e30e2c84e08c  
774  37f6489cfb4ee3ffda99e30e2c84e08c  
775  37f6489cfb4ee3ffda99e30e2c84e08c  

[460 rows x 7 columns]

out of the 778 rows in foursquare_df, 460 observations of those are not present in the data retuned by yelp API

In [58]:
#yelp_only = yelp_df[~yelp_df['name'].isin(foursquare_df['name'])]
yelp_only = yelp_df[~yelp_df['name'].isin(foursquare_df['name'])]
yelp_only


name              address     distance  \
0                   Aba - Austin  1011 S Congress Ave   471.988869   
1                Prim and Proper  1007 S Congress Ave   400.808357   
3     Summer House on Music Lane        1101 Music Ln   569.116145   
8              Austin Rotisserie     111 Congress Ave   850.685565   
9                    Lucky Robot  1303 S Congress Ave   748.499198   
...                          ...                  ...          ...   
1523          Taqueria La Morena     5419 Glissman Rd   605.458708   
1524              Native Grocery       705 Tillery St  1165.804443   
1525                 Panda Label     5323 Levander Lp   287.846224   
1526        Pueblo Viejo Tillery       641 Tillery St   982.714120   
1527               Solar Falafel         5305 Bolm Rd   952.034772   

           category_name  review_count  rating  longitude   latitude postcode  \
0          Mediterranean          1086     4.5 -97.748030  30.253570    78704   
1          Cocktail Bars            41     4.5 -97.747619  30.254225    78704   
3     Breakfast & Brunch           164     4.0 -97.747695  30.252618    78704   
8           Comfort Food           162     4.5 -97.743576  30.263536    78701   
9             Sushi Bars          1358     4.0 -97.749006  30.250851    78704   
...                  ...           ...     ...        ...        ...      ...   
1523             Mexican             5     4.0 -97.692612  30.254946    78702   
1524             Grocery            37     4.5 -97.704171  30.261115    78702   
1525             Chinese             0     0.0 -97.696014  30.250170    78702   
1526               Tacos             1     5.0 -97.705780  30.257818    78702   
1527             Falafel             0     0.0 -97.691116  30.257944    78721   

                            station_id price  
0     01eedbec47378a4407f5a3fc4352fdf3  None  
1     01eedbec47378a4407f5a3fc4352fdf3    $$  
3     01eedbec47378a4407f5a3fc4352fdf3    $$  
8     01eedbec47378a4407f5a3fc4352fdf3     $  
9     01eedbec47378a4407f5a3fc4352fdf3    $$  
...                                ...   ...  
1523  37f6489cfb4ee3ffda99e30e2c84e08c     $  
1524  37f6489cfb4ee3ffda99e30e2c84e08c  None  
1525  37f6489cfb4ee3ffda99e30e2c84e08c  None  
1526  37f6489cfb4ee3ffda99e30e2c84e08c  None  
1527  37f6489cfb4ee3ffda99e30e2c84e08c  None  

[884 rows x 11 columns]

Yelp has 884 unique observations that's not returned by the foursquare APi

In [ ]:
#compare both DataFrames based on multiple columns name and address
name_address_common_records = foursquare_df.merge(yelp_df, on=['name', 'address'], how='inner')
name_address_common_records


name                 address  category_id  \
0                    Taco Joint      134 E Riverside Dr        13145   
1                    Taco Joint      134 E Riverside Dr        13145   
2                    Taco Joint      134 E Riverside Dr        13145   
3                    Taco Joint      134 E Riverside Dr        13145   
4                    Taco Joint      134 E Riverside Dr        13145   
...                         ...                     ...          ...   
1516                Sawyer & Co  4827 E Cesar Chavez St        13049   
1517            De Nada Cantina  4715 E Cesar Chavez St        13009   
1518            De Nada Cantina  4715 E Cesar Chavez St        13009   
1519  Beto's Mexican Restaurant           3518 E 7th St        13303   
1520  Dan's Hamburgers - Austin        844 Airport Blvd        13031   

           category_name_x postcode_x  distance_x  distance_y category_name_y  \
0     Fast Food Restaurant      78704         354  370.641895         Mexican   
1     Fast Food Restaurant      78704         354  252.611181         Mexican   
2     Fast Food Restaurant      78704         354  395.004798         Mexican   
3     Fast Food Restaurant      78704         354  260.673302         Mexican   
4     Fast Food Restaurant      78704         354  437.020613         Mexican   
...                    ...        ...         ...         ...             ...   
1516                 Diner      78702         367  375.673199    Cajun/Creole   
1517          Cocktail Bar      78702         400  921.862744         Mexican   
1518          Cocktail Bar      78702         400  408.984337         Mexican   
1519    Mexican Restaurant      78702         651  674.326163         Mexican   
1520          Burger Joint      78702         808  807.452493         Burgers   

      review_count  rating  longitude   latitude postcode_y  
0              563     4.5 -97.745520  30.256370      78704  
1              563     4.5 -97.745520  30.256370      78704  
2              563     4.5 -97.745520  30.256370      78704  
3              563     4.5 -97.745520  30.256370      78704  
4              563     4.5 -97.745520  30.256370      78704  
...            ...     ...        ...        ...        ...  
1516           896     4.0 -97.701782  30.251333      78702  
1517           313     4.5 -97.702174  30.251447      78702  
1518           313     4.5 -97.702174  30.251447      78702  
1519            55     4.0 -97.702342  30.256849      78702  
1520           166     4.0 -97.693840  30.258469      78702  

[1521 rows x 13 columns]

In [ ]:
#combine/merge  foursquare_df and yelp_df
combined_df = pd.concat([foursquare_df, yelp_df])
combined_df


name                address  category_id  \
0          Sandy's Hamburgers  603 Barton Springs Rd      13031.0   
1     Freebirds World Burrito     515 S Congress Ave      13145.0   
2                  Taco Joint     134 E Riverside Dr      13145.0   
3                 Whataburger  601 Barton Springs Rd      13145.0   
4                  Aba Austin    1011 S Congress Ave      13000.0   
...                       ...                    ...          ...   
1524        B Cooper Barbecue          705 Gunter St          NaN   
1525           Native Grocery         705 Tillery St          NaN   
1526              Panda Label       5323 Levander Lp          NaN   
1527     Pueblo Viejo Tillery         641 Tillery St          NaN   
1528            Solar Falafel           5305 Bolm Rd          NaN   

             category_name postcode     distance  review_count  rating  \
0             Burger Joint    78704   213.000000           NaN     NaN   
1     Fast Food Restaurant    78704   270.000000           NaN     NaN   
2     Fast Food Restaurant    78704   354.000000           NaN     NaN   
3     Fast Food Restaurant    78704   172.000000           NaN     NaN   
4      Dining and Drinking    78704   426.000000           NaN     NaN   
...                    ...      ...          ...           ...     ...   
1524              Barbeque    78702   742.107530          43.0     5.0   
1525               Grocery    78702  1165.804443          37.0     4.5   
1526               Chinese    78702   287.846224           0.0     0.0   
1527                 Tacos    78702   982.714120           1.0     5.0   
1528               Falafel    78721   952.034772           0.0     0.0   

      longitude   latitude  
0           NaN        NaN  
1           NaN        NaN  
2           NaN        NaN  
3           NaN        NaN  
4           NaN        NaN  
...         ...        ...  
1524 -97.702690  30.257300  
1525 -97.704171  30.261115  
1526 -97.696014  30.250170  
1527 -97.705780  30.257818  
1528 -97.691116  30.257944  

[2307 rows x 10 columns]

In [ ]:
#get duplicate records based on the name and address columns
duplicate_records = combined_df[combined_df.duplicated(subset=['name', 'address'], keep=False)]
duplicate_records

name                address  category_id  \
0            Sandy's Hamburgers  603 Barton Springs Rd      13031.0   
1       Freebirds World Burrito     515 S Congress Ave      13145.0   
2                    Taco Joint     134 E Riverside Dr      13145.0   
3                   Whataburger  601 Barton Springs Rd      13145.0   
4                    Aba Austin    1011 S Congress Ave      13000.0   
...                         ...                    ...          ...   
1513             Kinda Tropical          3501 E 7th St          NaN   
1514                  Llama Kid    4620 E Cesar Chavez          NaN   
1518  Beto's Mexican Restaurant          3518 E 7th St          NaN   
1520  Dan's Hamburgers - Austin       844 Airport Blvd          NaN   
1523             Supper Friends       3012 Gonzales St          NaN   

              category_name postcode     distance  review_count  rating  \
0              Burger Joint    78704   213.000000           NaN     NaN   
1      Fast Food Restaurant    78704   270.000000           NaN     NaN   
2      Fast Food Restaurant    78704   354.000000           NaN     NaN   
3      Fast Food Restaurant    78704   172.000000           NaN     NaN   
4       Dining and Drinking    78704   426.000000           NaN     NaN   
...                     ...      ...          ...           ...     ...   
1513     Breakfast & Brunch    78702   694.151271         175.0     4.0   
1514               Peruvian    78702   474.740683         163.0     4.0   
1518                Mexican    78702   674.326163          55.0     4.0   
1520                Burgers    78702   807.452493         166.0     4.0   
1523  Venues & Event Spaces    78702  1172.130827          27.0     4.5   

      longitude   latitude  
0           NaN        NaN  
1           NaN        NaN  
2           NaN        NaN  
3           NaN        NaN  
4           NaN        NaN  
...         ...        ...  
1513 -97.703090  30.256550  
1514 -97.702924  30.252178  
1518 -97.702342  30.256849  
1520 -97.693840  30.258469  
1523 -97.706630  30.259630  

[2094 rows x 10 columns]

In [ ]:
# Drop duplicate records based on 'name'.
combined_df = combined_df.drop_duplicates(subset=['name'], keep='first')
combined_df


name                address  category_id  \
0          Sandy's Hamburgers  603 Barton Springs Rd      13031.0   
1     Freebirds World Burrito     515 S Congress Ave      13145.0   
2                  Taco Joint     134 E Riverside Dr      13145.0   
3                 Whataburger  601 Barton Springs Rd      13145.0   
4                  Aba Austin    1011 S Congress Ave      13000.0   
...                       ...                    ...          ...   
1524        B Cooper Barbecue          705 Gunter St          NaN   
1525           Native Grocery         705 Tillery St          NaN   
1526              Panda Label       5323 Levander Lp          NaN   
1527     Pueblo Viejo Tillery         641 Tillery St          NaN   
1528            Solar Falafel           5305 Bolm Rd          NaN   

             category_name postcode     distance  review_count  rating  \
0             Burger Joint    78704   213.000000           NaN     NaN   
1     Fast Food Restaurant    78704   270.000000           NaN     NaN   
2     Fast Food Restaurant    78704   354.000000           NaN     NaN   
3     Fast Food Restaurant    78704   172.000000           NaN     NaN   
4      Dining and Drinking    78704   426.000000           NaN     NaN   
...                    ...      ...          ...           ...     ...   
1524              Barbeque    78702   742.107530          43.0     5.0   
1525               Grocery    78702  1165.804443          37.0     4.5   
1526               Chinese    78702   287.846224           0.0     0.0   
1527                 Tacos    78702   982.714120           1.0     5.0   
1528               Falafel    78721   952.034772           0.0     0.0   

      longitude   latitude  
0           NaN        NaN  
1           NaN        NaN  
2           NaN        NaN  
3           NaN        NaN  
4           NaN        NaN  
...         ...        ...  
1524 -97.702690  30.257300  
1525 -97.704171  30.261115  
1526 -97.696014  30.250170  
1527 -97.705780  30.257818  
1528 -97.691116  30.257944  

[564 rows x 10 columns]

In [ ]:
# Drop duplicate records based on 'name' and 'address'
combined_df = combined_df.drop_duplicates(subset=['name'], keep='first')
combined_df

In [ ]:
# Reset the index of the merged data frame
combined_df = combined_df.reset_index(drop=True)
combined_df

name                address  category_id  \
0         Sandy's Hamburgers  603 Barton Springs Rd      13031.0   
1    Freebirds World Burrito     515 S Congress Ave      13145.0   
2                 Taco Joint     134 E Riverside Dr      13145.0   
3                Whataburger  601 Barton Springs Rd      13145.0   
4                 Aba Austin    1011 S Congress Ave      13000.0   
..                       ...                    ...          ...   
607       Taqueria La Morena       5419 Glissman Rd          NaN   
608              Panda Label       5323 Levander Lp          NaN   
609           Native Grocery         705 Tillery St          NaN   
610     Pueblo Viejo Tillery         641 Tillery St          NaN   
611            Solar Falafel           5305 Bolm Rd          NaN   

            category_name postcode     distance  review_count  rating  \
0            Burger Joint    78704   213.000000           NaN     NaN   
1    Fast Food Restaurant    78704   270.000000           NaN     NaN   
2    Fast Food Restaurant    78704   354.000000           NaN     NaN   
3    Fast Food Restaurant    78704   172.000000           NaN     NaN   
4     Dining and Drinking    78704   426.000000           NaN     NaN   
..                    ...      ...          ...           ...     ...   
607               Mexican      NaN   605.458708           5.0     4.0   
608               Chinese      NaN   287.846224           0.0     0.0   
609               Grocery      NaN  1165.804443          37.0     4.5   
610                 Tacos      NaN   982.714120           1.0     5.0   
611               Falafel      NaN   952.034772           0.0     0.0   

     longitude   latitude  
0          NaN        NaN  
1          NaN        NaN  
2          NaN        NaN  
3          NaN        NaN  
4          NaN        NaN  
..         ...        ...  
607 -97.692612  30.254946  
608 -97.696014  30.250170  
609 -97.704171  30.261115  
610 -97.705780  30.257818  
611 -97.691116  30.257944  

[612 rows x 10 columns]

Which API provided you with more complete data? Provide an explanation. 

The yelp API had a more complete complete data. It contains datas such as ratings, review counts, price, multiple addresses, and multiple phone numbers where applicable.

In [ ]:
yelp_df

name                  address     distance  \
0                   Aba - Austin      1011 S Congress Ave   471.988869   
1                Prim and Proper      1007 S Congress Ave   400.808357   
2                        Red Ash          303 Colorado St  1013.704865   
3     Summer House on Music Lane            1101 Music Ln   569.116145   
4                      Two Hands  1011 South Congress Ave   460.097583   
...                          ...                      ...          ...   
1524          Taqueria La Morena         5419 Glissman Rd   605.458708   
1525                 Panda Label         5323 Levander Lp   287.846224   
1526              Native Grocery           705 Tillery St  1165.804443   
1527        Pueblo Viejo Tillery           641 Tillery St   982.714120   
1528               Solar Falafel             5305 Bolm Rd   952.034772   

           category_name  review_count  rating  longitude   latitude  
0          Mediterranean          1081     4.5 -97.748030  30.253570  
1          Cocktail Bars            41     4.5 -97.747619  30.254225  
2                Italian          1200     4.5 -97.744672  30.265880  
3     Breakfast & Brunch           163     4.0 -97.747695  30.252618  
4                   Bars           328     4.0 -97.747924  30.253579  
...                  ...           ...     ...        ...        ...  
1524             Mexican             5     4.0 -97.692612  30.254946  
1525             Chinese             0     0.0 -97.696014  30.250170  
1526             Grocery            37     4.5 -97.704171  30.261115  
1527               Tacos             1     5.0 -97.705780  30.257818  
1528             Falafel             0     0.0 -97.691116  30.257944  

[1529 rows x 8 columns]

In [59]:
yelp_longitude_unique = yelp_df["longitude"].is_unique
yelp_longitude_unique

False

In [87]:
df = pd.merge(foursquare_df, yelp_df, on='station_id', how='inner')
#df.drop_duplicates(subset='station_id', inplace=True)
df

name_x               address_x  category_id  \
0                    Sandy's Hamburgers   603 Barton Springs Rd        13031   
1                    Sandy's Hamburgers   603 Barton Springs Rd        13031   
2                    Sandy's Hamburgers   603 Barton Springs Rd        13031   
3                    Sandy's Hamburgers   603 Barton Springs Rd        13031   
4                    Sandy's Hamburgers   603 Barton Springs Rd        13031   
...                                 ...                     ...          ...   
15226  Jacoby's Restaurant & Mercantile  3235 E Cesar Chavez St        13068   
15227  Jacoby's Restaurant & Mercantile  3235 E Cesar Chavez St        13068   
15228  Jacoby's Restaurant & Mercantile  3235 E Cesar Chavez St        13068   
15229  Jacoby's Restaurant & Mercantile  3235 E Cesar Chavez St        13068   
15230  Jacoby's Restaurant & Mercantile  3235 E Cesar Chavez St        13068   

           category_name_x postcode_x  distance_x  \
0             Burger Joint      78704         213   
1             Burger Joint      78704         213   
2             Burger Joint      78704         213   
3             Burger Joint      78704         213   
4             Burger Joint      78704         213   
...                    ...        ...         ...   
15226  American Restaurant      78702         891   
15227  American Restaurant      78702         891   
15228  American Restaurant      78702         891   
15229  American Restaurant      78702         891   
15230  American Restaurant      78702         891   

                             station_id                      name_y  \
0      01eedbec47378a4407f5a3fc4352fdf3                Aba - Austin   
1      01eedbec47378a4407f5a3fc4352fdf3             Prim and Proper   
2      01eedbec47378a4407f5a3fc4352fdf3                     Red Ash   
3      01eedbec47378a4407f5a3fc4352fdf3  Summer House on Music Lane   
4      01eedbec47378a4407f5a3fc4352fdf3                   Two Hands   
...                                 ...                         ...   
15226  37f6489cfb4ee3ffda99e30e2c84e08c          Taqueria La Morena   
15227  37f6489cfb4ee3ffda99e30e2c84e08c              Native Grocery   
15228  37f6489cfb4ee3ffda99e30e2c84e08c                 Panda Label   
15229  37f6489cfb4ee3ffda99e30e2c84e08c        Pueblo Viejo Tillery   
15230  37f6489cfb4ee3ffda99e30e2c84e08c               Solar Falafel   

                     address_y   distance_y     category_name_y  review_count  \
0          1011 S Congress Ave   471.988869       Mediterranean          1086   
1          1007 S Congress Ave   400.808357       Cocktail Bars            41   
2              303 Colorado St  1013.704865             Italian          1204   
3                1101 Music Ln   569.116145  Breakfast & Brunch           164   
4      1011 South Congress Ave   460.097583       Cocktail Bars           328   
...                        ...          ...                 ...           ...   
15226         5419 Glissman Rd   605.458708             Mexican             5   
15227           705 Tillery St  1165.804443             Grocery            37   
15228         5323 Levander Lp   287.846224             Chinese             0   
15229           641 Tillery St   982.714120               Tacos             1   
15230             5305 Bolm Rd   952.034772             Falafel             0   

       rating  longitude   latitude postcode_y price  
0         4.5 -97.748030  30.253570      78704  None  
1         4.5 -97.747619  30.254225      78704    $$  
2         4.5 -97.744672  30.265880      78701  $$$$  
3         4.0 -97.747695  30.252618      78704    $$  
4         4.0 -97.747924  30.253579      78704    $$  
...       ...        ...        ...        ...   ...  
15226     4.0 -97.692612  30.254946      78702     $  
15227     4.5 -97.704171  30.261115      78702  None  
15228     0.0 -97.696014  30.250170      78702  None  
15229     5.0 -97.705780  30.257818      78702  None  
1523

In [86]:
df.shape

(79, 17)

In [63]:
yelp_foursquare_df = df.copy()
yelp_foursquare_df['name'] = df['name_x'].fillna(df['name_y'])
yelp_foursquare_df['address'] = df['address_x'].fillna(df['address_y'])
yelp_foursquare_df['category_name'] = df['category_name_x'].fillna(df['category_name_y'])
yelp_foursquare_df['postcode'] = df['postcode_x'].fillna(df['postcode_y'])
yelp_foursquare_df['distance'] = df['distance_x'].fillna(df['distance_y'])
yelp_foursquare_df

name_x              address_x  category_id  \
0        Sandy's Hamburgers  603 Barton Springs Rd        13031   
200          Torchy’S Tacos      2801 Guadalupe St        13145   
400       Pluckers Wing Bar     2222 Rio Grande St        13022   
600                24 Diner       600 N Lamar Blvd        13049   
800              Aba Austin    1011 S Congress Ave        13000   
...                     ...                    ...          ...   
14231       Kemuri Tatsu-ya          2713 E 2nd St        13026   
14431               Irene's           506 West Ave        13049   
14631  Modern Market Eatery       401 Congress Ave        13145   
14831            Chi'Lantro       823 Congress Ave        13145   
15031   Justine's Brasserie          4710 E 5th St        13148   

            category_name_x postcode_x  distance_x  \
0              Burger Joint      78704         213   
200    Fast Food Restaurant      78705         238   
400              Sports Bar      78705         193   
600                   Diner      78703         150   
800     Dining and Drinking      78704         114   
...                     ...        ...         ...   
14231             BBQ Joint      78702         381   
14431                 Diner      78701          77   
14631  Fast Food Restaurant      78701          45   
14831  Fast Food Restaurant      78701         198   
15031     French Restaurant      78702         264   

                             station_id                       name_y  \
0      01eedbec47378a4407f5a3fc4352fdf3                 Aba - Austin   
200    57eb8e5d401aeabe67ff95e6f1f1f999                      Daiboku   
400    ed95d8e12ad7c2ec3ff0b7dd0f4673f2                      Acre 41   
600    8723bfa08ec83b133f6a9aeecd075f51                    Qi Austin   
800    568256d098257d6e8433513638f5257b                 Aba - Austin   
...                                 ...                          ...   
14231  fe9c5e51f706fdd8981ed72ebf05eab4                         Ezov   
14431  3c8c0510f41de2f9d61a1d3ed5993790                    Qi Austin   
14631  15c906ebd3955b31efa9a1dd4de8f950  Moonshine Patio Bar & Grill   
14831  49abfbfe3cddcd1574965bef3772db3d                Higher Ground   
15031  37f6489cfb4ee3ffda99e30e2c84e08c                  Sawyer & Co   

                    address_y  distance_y  ... rating  longitude   latitude  \
0         1011 S Congress Ave  471.988869  ...    4.5 -97.748030  30.253570   
200             609 W 29th St  257.731190  ...    4.5 -97.743651  30.295438   
400       1901 San Antonio St  513.348715  ...    4.0 -97.742910  30.282230   
600           835 West 6th St  165.638817  ...    4.5 -97.752436  30.270623   
800       1011 S Congress Ave  162.170770  ...    4.5 -97.748030  30.253570   
...                       ...         ...  ...    ...        ...        ...   
14231  2708 E Cesar Chavez St  458.686411  ...    4.5 -97.713519  30.253460   
14431         835 West 6th St  187.656838  ...    4.5 -97.752436  30.270623   
14631        303 Red River St  621.196233  ...    4.5 -97.738077  30.263754   
14831        720 Congress Ave  327.857155  ...    4.0 -97.742400  30.269780   
15031  4827 E Cesar Chavez St  375.673199  ...    4.0 -97.701782  30.251333   

       postcode_y  price                  name                address  \
0           78704   None    Sandy's Hamburgers  603 Barton Springs Rd   
200         78705   None        Torchy’S Tacos      2801 Guadalupe St   
400         78705    $$$     Pluckers Wing Bar     2222 Rio Grande St   
600         78703     $$              24 Diner       600 N Lamar Blvd   
800         78704   None            Aba Austin    1011 S Congress Ave   
...           ...    ...                   ...                    ...   
14231       78702   None       Kemuri Tatsu-ya          2713 E 2nd St   
14431       78703     $$               Irene's           506 West Ave   
14631       78701     $$  Modern Market Eatery       401 Congress Ave   
14831       78701     $

In [64]:
yelp_foursquare_df.drop(['address_y', 'name_y', 'category_name_y', 'postcode_y', 'distance_y'], axis=1, inplace=True)

yelp_foursquare_df


name_x              address_x  category_id  \
0        Sandy's Hamburgers  603 Barton Springs Rd        13031   
200          Torchy’S Tacos      2801 Guadalupe St        13145   
400       Pluckers Wing Bar     2222 Rio Grande St        13022   
600                24 Diner       600 N Lamar Blvd        13049   
800              Aba Austin    1011 S Congress Ave        13000   
...                     ...                    ...          ...   
14231       Kemuri Tatsu-ya          2713 E 2nd St        13026   
14431               Irene's           506 West Ave        13049   
14631  Modern Market Eatery       401 Congress Ave        13145   
14831            Chi'Lantro       823 Congress Ave        13145   
15031   Justine's Brasserie          4710 E 5th St        13148   

            category_name_x postcode_x  distance_x  \
0              Burger Joint      78704         213   
200    Fast Food Restaurant      78705         238   
400              Sports Bar      78705         193   
600                   Diner      78703         150   
800     Dining and Drinking      78704         114   
...                     ...        ...         ...   
14231             BBQ Joint      78702         381   
14431                 Diner      78701          77   
14631  Fast Food Restaurant      78701          45   
14831  Fast Food Restaurant      78701         198   
15031     French Restaurant      78702         264   

                             station_id  review_count  rating  longitude  \
0      01eedbec47378a4407f5a3fc4352fdf3          1086     4.5 -97.748030   
200    57eb8e5d401aeabe67ff95e6f1f1f999            15     4.5 -97.743651   
400    ed95d8e12ad7c2ec3ff0b7dd0f4673f2           172     4.0 -97.742910   
600    8723bfa08ec83b133f6a9aeecd075f51           547     4.5 -97.752436   
800    568256d098257d6e8433513638f5257b          1086     4.5 -97.748030   
...                                 ...           ...     ...        ...   
14231  fe9c5e51f706fdd8981ed72ebf05eab4            51     4.5 -97.713519   
14431  3c8c0510f41de2f9d61a1d3ed5993790           547     4.5 -97.752436   
14631  15c906ebd3955b31efa9a1dd4de8f950          5773     4.5 -97.738077   
14831  49abfbfe3cddcd1574965bef3772db3d           166     4.0 -97.742400   
15031  37f6489cfb4ee3ffda99e30e2c84e08c           895     4.0 -97.701782   

        latitude price                  name                address  \
0      30.253570  None    Sandy's Hamburgers  603 Barton Springs Rd   
200    30.295438  None        Torchy’S Tacos      2801 Guadalupe St   
400    30.282230   $$$     Pluckers Wing Bar     2222 Rio Grande St   
600    30.270623    $$              24 Diner       600 N Lamar Blvd   
800    30.253570  None            Aba Austin    1011 S Congress Ave   
...          ...   ...                   ...                    ...   
14231  30.253460  None       Kemuri Tatsu-ya          2713 E 2nd St   
14431  30.270623    $$               Irene's           506 West Ave   
14631  30.263754    $$  Modern Market Eatery       401 Congress Ave   
14831  30.269780    $$            Chi'Lantro       823 Congress Ave   
15031  30.251333    $$   Justine's Brasserie          4710 E 5th St   

              category_name postcode  distance  
0              Burger Joint    78704       213  
200    Fast Food Restaurant    78705       238  
400              Sports Bar    78705       193  
600                   Diner    78703       150  
800     Dining and Drinking    78704       114  
...                     ...      ...       ...  
14231             BBQ Joint    78702       381  
14431                 Diner    78701        77  
14631  Fast Food Restaurant    78701        45  
14831  Fast Food Restaurant    78701       198  
15031     French Restaurant    78702       264  

[79 rows x 17 columns]

In [65]:
yelp_foursquare_df.drop(['address_x', 'name_x', 'category_name_x', 'postcode_x', 'distance_x'], axis=1, inplace=True)

yelp_foursquare_df

category_id                        station_id  review_count  rating  \
0            13031  01eedbec47378a4407f5a3fc4352fdf3          1086     4.5   
200          13145  57eb8e5d401aeabe67ff95e6f1f1f999            15     4.5   
400          13022  ed95d8e12ad7c2ec3ff0b7dd0f4673f2           172     4.0   
600          13049  8723bfa08ec83b133f6a9aeecd075f51           547     4.5   
800          13000  568256d098257d6e8433513638f5257b          1086     4.5   
...            ...                               ...           ...     ...   
14231        13026  fe9c5e51f706fdd8981ed72ebf05eab4            51     4.5   
14431        13049  3c8c0510f41de2f9d61a1d3ed5993790           547     4.5   
14631        13145  15c906ebd3955b31efa9a1dd4de8f950          5773     4.5   
14831        13145  49abfbfe3cddcd1574965bef3772db3d           166     4.0   
15031        13148  37f6489cfb4ee3ffda99e30e2c84e08c           895     4.0   

       longitude   latitude price                  name  \
0     -97.748030  30.253570  None    Sandy's Hamburgers   
200   -97.743651  30.295438  None        Torchy’S Tacos   
400   -97.742910  30.282230   $$$     Pluckers Wing Bar   
600   -97.752436  30.270623    $$              24 Diner   
800   -97.748030  30.253570  None            Aba Austin   
...          ...        ...   ...                   ...   
14231 -97.713519  30.253460  None       Kemuri Tatsu-ya   
14431 -97.752436  30.270623    $$               Irene's   
14631 -97.738077  30.263754    $$  Modern Market Eatery   
14831 -97.742400  30.269780    $$            Chi'Lantro   
15031 -97.701782  30.251333    $$   Justine's Brasserie   

                     address         category_name postcode  distance  
0      603 Barton Springs Rd          Burger Joint    78704       213  
200        2801 Guadalupe St  Fast Food Restaurant    78705       238  
400       2222 Rio Grande St            Sports Bar    78705       193  
600         600 N Lamar Blvd                 Diner    78703       150  
800      1011 S Congress Ave   Dining and Drinking    78704       114  
...                      ...                   ...      ...       ...  
14231          2713 E 2nd St             BBQ Joint    78702       381  
14431           506 West Ave                 Diner    78701        77  
14631       401 Congress Ave  Fast Food Restaurant    78701        45  
14831       823 Congress Ave  Fast Food Restaurant    78701       198  
15031          4710 E 5th St     French Restaurant    78702       264  

[79 rows x 12 columns]

In [70]:
yelp_foursquare_df = yelp_foursquare_df[['name', 'station_id', 'category_id', 'category_name', 'review_count', 'distance', 'rating', 'address', 'postcode']]
yelp_foursquare_df

name                        station_id  category_id  \
0        Sandy's Hamburgers  01eedbec47378a4407f5a3fc4352fdf3        13031   
200          Torchy’S Tacos  57eb8e5d401aeabe67ff95e6f1f1f999        13145   
400       Pluckers Wing Bar  ed95d8e12ad7c2ec3ff0b7dd0f4673f2        13022   
600                24 Diner  8723bfa08ec83b133f6a9aeecd075f51        13049   
800              Aba Austin  568256d098257d6e8433513638f5257b        13000   
...                     ...                               ...          ...   
14231       Kemuri Tatsu-ya  fe9c5e51f706fdd8981ed72ebf05eab4        13026   
14431               Irene's  3c8c0510f41de2f9d61a1d3ed5993790        13049   
14631  Modern Market Eatery  15c906ebd3955b31efa9a1dd4de8f950        13145   
14831            Chi'Lantro  49abfbfe3cddcd1574965bef3772db3d        13145   
15031   Justine's Brasserie  37f6489cfb4ee3ffda99e30e2c84e08c        13148   

              category_name  review_count  distance  rating  \
0              Burger Joint          1086       213     4.5   
200    Fast Food Restaurant            15       238     4.5   
400              Sports Bar           172       193     4.0   
600                   Diner           547       150     4.5   
800     Dining and Drinking          1086       114     4.5   
...                     ...           ...       ...     ...   
14231             BBQ Joint            51       381     4.5   
14431                 Diner           547        77     4.5   
14631  Fast Food Restaurant          5773        45     4.5   
14831  Fast Food Restaurant           166       198     4.0   
15031     French Restaurant           895       264     4.0   

                     address postcode  
0      603 Barton Springs Rd    78704  
200        2801 Guadalupe St    78705  
400       2222 Rio Grande St    78705  
600         600 N Lamar Blvd    78703  
800      1011 S Congress Ave    78704  
...                      ...      ...  
14231          2713 E 2nd St    78702  
14431           506 West Ave    78701  
14631       401 Congress Ave    78701  
14831       823 Congress Ave    78701  
15031          4710 E 5th St    78702  

[79 rows x 9 columns]

In [74]:
yelp_foursquare_df

name                        station_id  category_id  \
0        Sandy's Hamburgers  01eedbec47378a4407f5a3fc4352fdf3        13031   
200          Torchy’S Tacos  57eb8e5d401aeabe67ff95e6f1f1f999        13145   
400       Pluckers Wing Bar  ed95d8e12ad7c2ec3ff0b7dd0f4673f2        13022   
600                24 Diner  8723bfa08ec83b133f6a9aeecd075f51        13049   
800              Aba Austin  568256d098257d6e8433513638f5257b        13000   
...                     ...                               ...          ...   
14231       Kemuri Tatsu-ya  fe9c5e51f706fdd8981ed72ebf05eab4        13026   
14431               Irene's  3c8c0510f41de2f9d61a1d3ed5993790        13049   
14631  Modern Market Eatery  15c906ebd3955b31efa9a1dd4de8f950        13145   
14831            Chi'Lantro  49abfbfe3cddcd1574965bef3772db3d        13145   
15031   Justine's Brasserie  37f6489cfb4ee3ffda99e30e2c84e08c        13148   

              category_name  review_count  distance  rating  \
0              Burger Joint          1086       213     4.5   
200    Fast Food Restaurant            15       238     4.5   
400              Sports Bar           172       193     4.0   
600                   Diner           547       150     4.5   
800     Dining and Drinking          1086       114     4.5   
...                     ...           ...       ...     ...   
14231             BBQ Joint            51       381     4.5   
14431                 Diner           547        77     4.5   
14631  Fast Food Restaurant          5773        45     4.5   
14831  Fast Food Restaurant           166       198     4.0   
15031     French Restaurant           895       264     4.0   

                     address postcode  
0      603 Barton Springs Rd    78704  
200        2801 Guadalupe St    78705  
400       2222 Rio Grande St    78705  
600         600 N Lamar Blvd    78703  
800      1011 S Congress Ave    78704  
...                      ...      ...  
14231          2713 E 2nd St    78702  
14431           506 West Ave    78701  
14631       401 Congress Ave    78701  
14831       823 Congress Ave    78701  
15031          4710 E 5th St    78702  

[79 rows x 9 columns]

In [78]:
yelp_foursquare_df_unique = yelp_foursquare_df["category_id"].is_unique
yelp_foursquare_df_unique

False

In [79]:
yelp_foursquare_df.head

<bound method NDFrame.head of                        name                        station_id  category_id  \
0        Sandy's Hamburgers  01eedbec47378a4407f5a3fc4352fdf3        13031   
200          Torchy’S Tacos  57eb8e5d401aeabe67ff95e6f1f1f999        13145   
400       Pluckers Wing Bar  ed95d8e12ad7c2ec3ff0b7dd0f4673f2        13022   
600                24 Diner  8723bfa08ec83b133f6a9aeecd075f51        13049   
800              Aba Austin  568256d098257d6e8433513638f5257b        13000   
...                     ...                               ...          ...   
14231       Kemuri Tatsu-ya  fe9c5e51f706fdd8981ed72ebf05eab4        13026   
14431               Irene's  3c8c0510f41de2f9d61a1d3ed5993790        13049   
14631  Modern Market Eatery  15c906ebd3955b31efa9a1dd4de8f950        13145   
14831            Chi'Lantro  49abfbfe3cddcd1574965bef3772db3d        13145   
15031   Justine's Brasserie  37f6489cfb4ee3ffda99e30e2c84e08c        13148   

              category_name  revi

In [80]:
yelp_foursquare_df.reset_index(drop=True, inplace=True)


In [81]:
yelp_foursquare_df

name                        station_id  category_id  \
0     Sandy's Hamburgers  01eedbec47378a4407f5a3fc4352fdf3        13031   
1         Torchy’S Tacos  57eb8e5d401aeabe67ff95e6f1f1f999        13145   
2      Pluckers Wing Bar  ed95d8e12ad7c2ec3ff0b7dd0f4673f2        13022   
3               24 Diner  8723bfa08ec83b133f6a9aeecd075f51        13049   
4             Aba Austin  568256d098257d6e8433513638f5257b        13000   
..                   ...                               ...          ...   
74       Kemuri Tatsu-ya  fe9c5e51f706fdd8981ed72ebf05eab4        13026   
75               Irene's  3c8c0510f41de2f9d61a1d3ed5993790        13049   
76  Modern Market Eatery  15c906ebd3955b31efa9a1dd4de8f950        13145   
77            Chi'Lantro  49abfbfe3cddcd1574965bef3772db3d        13145   
78   Justine's Brasserie  37f6489cfb4ee3ffda99e30e2c84e08c        13148   

           category_name  review_count  distance  rating  \
0           Burger Joint          1086       213     4.5   
1   Fast Food Restaurant            15       238     4.5   
2             Sports Bar           172       193     4.0   
3                  Diner           547       150     4.5   
4    Dining and Drinking          1086       114     4.5   
..                   ...           ...       ...     ...   
74             BBQ Joint            51       381     4.5   
75                 Diner           547        77     4.5   
76  Fast Food Restaurant          5773        45     4.5   
77  Fast Food Restaurant           166       198     4.0   
78     French Restaurant           895       264     4.0   

                  address postcode  
0   603 Barton Springs Rd    78704  
1       2801 Guadalupe St    78705  
2      2222 Rio Grande St    78705  
3        600 N Lamar Blvd    78703  
4     1011 S Congress Ave    78704  
..                    ...      ...  
74          2713 E 2nd St    78702  
75           506 West Ave    78701  
76       401 Congress Ave    78701  
77       823 Congress Ave    78701  
78          4710 E 5th St    78702  

[79 rows x 9 columns]

In [85]:
yelp_foursquare_df.to_csv("yelp_foursquare.csv", index = False)

In [84]:
#get duplicate records based on the name and address columns
duplicate_records = yelp_foursquare_df[yelp_foursquare_df.duplicated(subset=['name'], keep=False)]
duplicate_records
duplicate_records.shape

(32, 9)

In [93]:
grouped_df = yelp_foursquare_df.groupby('category_name').size().reset_index(name='count')
grouped_df.shape
grouped_df.info

<bound method DataFrame.info of                 category_name  count
0         American Restaurant     17
1                   BBQ Joint      5
2        Brazilian Restaurant      1
3                Burger Joint      5
4                        Café      5
5                Cocktail Bar      3
6        Colombian Restaurant      1
7                        Deli      1
8                       Diner      2
9         Dining and Drinking      5
10       Fast Food Restaurant      7
11          French Restaurant      1
12        Fried Chicken Joint      1
13        Hawaiian Restaurant      1
14                      Hotel      1
15         Israeli Restaurant      1
16         Italian Restaurant      1
17  Japanese Curry Restaurant      1
18  Latin American Restaurant      1
19         Mexican Restaurant      1
20                   Pizzeria      5
21              Sandwich Spot      3
22                 Sports Bar      3
23                 Steakhouse      2
24           Sushi Restaurant      2
25    

Get the top 10 restaurants according to their rating

In [82]:
top_10_restaurants = yelp_foursquare_df.sort_values(by='rating', ascending=False).head(10)
top_10_restaurants


name  \
0                                 Sandy's Hamburgers   
68                 Fogo de Chão Brazilian Steakhouse   
31                                  Home Slice Pizza   
67                                            Suerte   
33                              Super Burrito on 5th   
34                                 True Food Kitchen   
35  Iron Cactus Mexican Restaurant and Margarita Bar   
36                                 Thundercloud Subs   
37                            Terry Black's Barbecue   
66                                        Fareground   

                          station_id  category_id         category_name  \
0   01eedbec47378a4407f5a3fc4352fdf3        13031          Burger Joint   
68  bcedb3083b605964fa6f38bfb9e7004a        13079  Brazilian Restaurant   
31  8c8d0165291611b45159d72776264731        13064              Pizzeria   
67  feeb818fb554174880fb7f8b05dcc126        13303    Mexican Restaurant   
33  10671dfd7f66bf800de1c54e65948b67        13145  Fast Food Restaurant   
34  1cd4d71de52d29c3fdcd8c19db997075        13068   American Restaurant   
35  bc2824413e748667b80d555de755153b        13031          Burger Joint   
36  2351f7a493af99ec651013c7ff3f2553        13145  Fast Food Restaurant   
37  1130dd8e9a9b8e99a8d8165cca951e48        13026             BBQ Joint   
66  4cddd031571f32ffc4ae49d78076dc9c        13000   Dining and Drinking   

    review_count  distance  rating                 address postcode  
0           1086       213     4.5   603 Barton Springs Rd    78704  
68          5773        35     4.5        200 Congress Ave    78701  
31          1086        42     4.5     1415 S Congress Ave    78704  
67           954       112     4.5           1800 E 6th St    78702  
33           547        75     4.5            817 W 5th St    78703  
34           547       252     4.5            222 West Ave    78701  
35          5773         7     4.5          606 Trinity St    78701  
36           547       308     4.5           903 W 12th St    78703  
37          1086       144     4.5  1003 Barton Springs Rd    78704  
66          5773        91     4.5            Congress Ave    78701